crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -3958,6 +3980,11 @@ impl<S> StateMachineApplyContext<'_, S> {
                 .await
                 .map_err(Error::Storage)?;
         }
+        if events_length != 0 {
+            JournalEventsTable::delete_journal_events(self.storage, invocation_id, events_length)
+                .await
+                .map_err(Error::Storage)?;
+        }
```

Comment:
Nice. I imagine a future where the journal is dropped while we keep events for a little longer for log/auditing. Assuming that events are extended to include other event types for timeline tracking.